StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1120CodeFixProvider.cs

```diff
@@ -0,0 +1,107 @@
+    using Microsoft.CodeAnalysis.CodeActions;
+    using Microsoft.CodeAnalysis.CSharp;
+    using StyleCop.Analyzers.Helpers;
+
+    /// <summary>
+    /// Implements a code fix for <see cref="SA1120CommentsMustContainText"/>.
+    /// </summary>
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = nameof(SA1120CodeFixProvider))]
+    [Shared]
+    public class SA1120CodeFixProvider : CodeFixProvider
+    {
+        private static readonly ImmutableArray<string> FixableDiagnostics =
+            ImmutableArray.Create(SA1120CommentsMustContainText.DiagnosticId);
+
+        /// <inheritdoc/>
+        public override ImmutableArray<string> FixableDiagnosticIds => FixableDiagnostics;
+
+        /// <inheritdoc/>
+        public override FixAllProvider GetFixAllProvider()
+        {
+            return WellKnownFixAllProviders.BatchFixer;
+        }
+
+        /// <inheritdoc/>
+        public override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            foreach (var diagnostic in context.Diagnostics.Where(d => FixableDiagnostics.Contains(d.Id)))
+            {
+                context.RegisterCodeFix(CodeAction.Create(ReadabilityResources.SA1120CodeFix, token => GetTransformedDocumentAsync(context.Document, diagnostic, token)), diagnostic);
+            }
+
+            return SpecializedTasks.CompletedTask;
+        }
+
+        private static async Task<Document> GetTransformedDocumentAsync(Document document, Diagnostic diagnostic, CancellationToken cancellationToken)
+        {
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var node = root.FindTrivia(diagnostic.Location.SourceSpan.Start, true);
+
+            int diagnosticIndex = 0;
+            var triviaList = TriviaHelper.GetContainingTriviaList(node, out diagnosticIndex);
+
+            var nodesToRemove = new List<SyntaxTrivia>();
+            nodesToRemove.Add(node);
+
+            // If the comment is the first non-whitespace item in the trivia list, we are going to remove all of the leading whitespace.
+            var firstIndex = TriviaHelper.IndexOfFirstNonWhitespaceTrivia(triviaList);
+
+            bool removedLeadingWhitespace = false;
+
+            if (firstIndex == diagnosticIndex)
+            {
+                removedLeadingWhitespace = true;
+                for (int i = diagnosticIndex; i >= 0; i--)
+                {
+                    var start = diagnostic.Location.SourceSpan.Start - (triviaList[diagnosticIndex].SpanStart - triviaList[i].SpanStart);
+                    var prevNode = root.FindTrivia(start, true);
+                    nodesToRemove.Add(prevNode);
+                }
+            }
+
+            var lastIndex = TriviaHelper.IndexOfTrailingWhitespace(triviaList);
+
+            // If the comment is the last non-whitespace item in the trivia list and it also wasn't the first non-whitespace item in the trivia list, then we remove the trailing whitespace.
```

Comment:
:bulb: Remove at most one token following the comment, and do so *only* if that token is an end-of-line token.